lib/customer.rb

```diff
@@ -0,0 +1,44 @@
+require 'csv'
+
+class Customer
+  attr_reader :id
+  attr_accessor :email, :address
+
+  @@customers = []
+  # Create constructor method for each .new instance of Customer
+  def initialize(id, email, address)
+    @id = id.to_i
+    @email = email
+    @address = address
+  end
+
+  # Method to add each Customer.new to array of Customer instances
+  def add_to_customers
+    @@customers << self
+  end
+
+  # Customer class method to list all of Customer instances
+  def self.all
+    @@customers
+  end
+
+  # Customer class method to find instance of Customer by id
+  def self.find(id)
+    found_customer = nil
+
+    Customer.all.each do |customer|
+      if customer.id == id
+        found_customer = customer
+      end
+    end
+
+    return found_customer
+  end
+end
+
+# Open CSV of Customer data and read each into instance of Customer 
+CSV.open("data/customers.csv",'r').each do |line|
+  address = {street: line[2], city: line[3], state: line[4], zip: line[5]}
+  cust = Customer.new(line[0], line[1], address)
+  cust.add_to_customers
```

Comment:
This works, but it's not exactly what we were expecting from this assignment. Our intention was that this work would be done in `.all`, rather than when the file is loaded.

Ultimately you end up having a trade-off. If you put this work in `.all`, you have to reload all the data from disk every time `.all` is called. However, the advantage is that if that data changes while your program is running (maybe another program modifies the file), then your program will always immediately get the fresh data. It's also a little more complex, since you're using that class variable.

My solution would be something in the middle: put this logic in a class method called `.reload`, and have `.all` call `.reload` if the data hasn't been loaded yet. Then if you need to refresh from disk you can.